README.md

```diff
@@ -1,59 +1,162 @@
+  - [Load travel-sample bucket in Couchbase Capella](https://docs.couchbase.com/cloud/clusters/data-service/import-data-documents.html#import-sample-data)
 
-## Install Dependencies 
+## App Setup
 
+We will walk through the different steps required to get the application running.
 
-```sh
-mvn package
+### Cloning Repo
+
+```shell
+git clone https://github.com/couchbase-examples/java-springboot-quickstart
 ```
-> Note: Maven packages auto restore when building the project in IntelliJ IDEA or Eclipse depending on IDE configuration.
 
-### Database Server Configuration
+### Install Dependencies
+
+The dependencies for the application are specified in the `pom.xml` file in the source folder. Dependencies can be installed through `mvn` the default package manager for Java.
+
+```
+mvn clean install -DskipTests=true
+```
+
+Note: The `-DskipTests=true` option is used to skip the tests. The tests require the application to be running.
+
+Note: The application is tested with Java 17. If you are using a different version of Java, please update the `pom.xml` file accordingly.
+
+### Setup Database Configuration
 
-All configuration for communication with the database is stored in the `/src/main/resources/application.properties` file.  This includes the connection string, username, and password.  The default username is assumed to be `Administrator` and the default password is assumed to be `password`.  If these are different in your environment you will need to change them before running the application.
+To learn more about connecting to your Capella cluster, please follow the [instructions](https://docs.couchbase.com/cloud/get-started/connect.html).
 
-### Dependency Injection via DBSetupRunner class 
+Specifically, you need to do the following:
 
-The quickstart code provides a CommandLineRunner called DBSetupRunner in the runners folder that wires up the Bucket and Cluster objects for dependency injection.  This runner also creates the bucket, collection, scope, and indexes for the tutorial to run properly automatically when the application 
+- Create the [database credentials](https://docs.couchbase.com/cloud/clusters/manage-database-users.html) to access the travel-sample bucket (Read and Write) used in the application.
+- [Allow access](https://docs.couchbase.com/cloud/clusters/allow-ip-address.html) to the Cluster from the IP on which the application is running.
+
+All configuration for communication with the database is read from the environment variables. We have provided a convenience feature in this quickstart to read the environment variables from a local file, `application.properties` in the `src/main/resources` folder.
+
+```properties
+spring.couchbase.bootstrap-hosts=#######
+spring.couchbase.bucket.name=travel-sample
+spring.couchbase.bucket.user=#######
+spring.couchbase.bucket.password=#######
+spring.mvc.pathmatch.matching-strategy=ANT_PATH_MATCHER
```

Comment:
We are actually using this matching routes like airline/{id}. 
There are other matchers as well like regex matcher,prefix matcher, full path matcher.

without explicitly mentioning this property, we get the following error:
```
Caused by: java.lang.NullPointerException: Cannot invoke "org.springframework.web.servlet.mvc.condition.PatternsRequestCondition.getPatterns()" because "this.condition" is null
```